master/internal/resourcemanagers/agent_resource_manager.go

```diff
@@ -156,3 +175,180 @@ func (a *agentResourceManager) aggregateTaskSummaries(
+	instanceType := na
+
+	if pool.Provider != nil {
+		if pool.Provider.AWS != nil {
+			poolType = resourcepoolv1.ResourcePoolType_RESOURCE_POOL_TYPE_AWS
+			preemptible = pool.Provider.AWS.SpotEnabled
+			location = pool.Provider.AWS.Region
+			imageID = pool.Provider.AWS.ImageID
+			instanceType = string(pool.Provider.AWS.InstanceType)
+		}
+		if pool.Provider.GCP != nil {
+			poolType = resourcepoolv1.ResourcePoolType_RESOURCE_POOL_TYPE_GCP
+			preemptible = pool.Provider.GCP.InstanceType.Preemptible
+			location = pool.Provider.GCP.Zone
+			imageID = pool.Provider.GCP.BootDiskSourceImage
+
+			instanceTypeStringBuilder := strings.Builder{}
+			instanceTypeStringBuilder.WriteString(pool.Provider.GCP.InstanceType.MachineType)
+			instanceTypeStringBuilder.WriteString(", ")
+			instanceTypeStringBuilder.WriteString(string(pool.Provider.GCP.InstanceType.GPUNum))
+			instanceTypeStringBuilder.WriteString("x")
+			instanceTypeStringBuilder.WriteString(pool.Provider.GCP.InstanceType.GPUType)
+			instanceType = instanceTypeStringBuilder.String()
+		}
+	}
+
+	var schedulerType resourcepoolv1.SchedulerType
+	if pool.Scheduler == nil {
+		ctx.Log().Errorf("scheduler is not present in config")
+		return &resourcepoolv1.ResourcePool{}, err
+	}
+
+	if pool.Scheduler.FairShare != nil {
+		schedulerType = resourcepoolv1.SchedulerType_SCHEDULER_TYPE_FAIR_SHARE
+	}
+	if pool.Scheduler.Priority != nil {
+		schedulerType = resourcepoolv1.SchedulerType_SCHEDULER_TYPE_PRIORITY
+	}
+	if pool.Scheduler.RoundRobin != nil {
+		schedulerType = resourcepoolv1.SchedulerType_SCHEDULER_TYPE_ROUND_ROBIN
+	}
+
+	resp := &resourcepoolv1.ResourcePool{
+		Name:                         pool.PoolName,
+		Description:                  pool.Description,
+		Type:                         poolType,
+		DefaultCpuPool:               a.config.DefaultCPUResourcePool == poolName,
+		DefaultGpuPool:               a.config.DefaultGPUResourcePool == poolName,
+		Preemptible:                  preemptible,
+		CpuContainerCapacityPerAgent: int32(pool.MaxCPUContainersPerAgent),
+		SchedulerType:                schedulerType,
+		Location:                     location,
+		ImageId:                      imageID,
+		InstanceType:                 instanceType,
+		Details:                      &resourcepoolv1.ResourcePoolDetail{},
+	}
+	if pool.Provider != nil {
+		resp.MinAgents = int32(pool.Provider.MinInstances)
+		resp.MaxAgents = int32(pool.Provider.MaxInstances)
+		resp.MasterUrl = pool.Provider.MasterURL
+		resp.MasterCertName = pool.Provider.MasterCertName
+		resp.StartupScript = pool.Provider.StartupScript
+		resp.ContainerStartupScript = pool.Provider.ContainerStartupScript
+		resp.AgentDockerNetwork = pool.Provider.AgentDockerNetwork
+		resp.AgentDockerRuntime = pool.Provider.AgentDockerRuntime
+		resp.AgentDockerImage = pool.Provider.AgentDockerImage
+		resp.AgentFluentImage = pool.Provider.AgentFluentImage
+		resp.MaxIdleAgentPeriod = time.Duration(pool.Provider.MaxIdleAgentPeriod).String()
```

Comment:
question: how does the user specify this? I'm thinking if this is returned in seconds it's easier to operate on

Reply:
same as with other durations.

Reply:
Same answer as above. I don't see a case where the user would want to operate on this value other than displaying it, so I would keep it as a string

Reply:
Updated to be float seconds